### ISSUE TITLE
Document how to use .NET Core 3.0 SDK to build an app for the 2.2 runtime

### ISSUE THREAD

**Issue by TomasHubelbauer** [dotnet/docs#14572]
Hi, I don't believe the specified page in its current form answers the following (and IMO pretty basic) question: I am using a .NET Core 3.0  RC1 SDK on my dev machine and am looking to deploy to a Linode node which has the .NET Core 2.2 runtime installed. We know the SDK uses the latest installed SDK version which also (I think?) means the project templates for that SDK version will be used with `dotnet new`. The `mvc` template for example will add NUGet package dependencies to packages targeting .NET Core 3.0 RC1 (because that's my SDK version). They will of course also set the `TargetFramework` to `netcoreapp3.0`. Now I want to build for the .NET 2.2 runtime. I can't just change the TFM to `netcoreapp2.0` because the NuGet packages are still pointing to the 3.0 RC1 versions. Now I could go ahead and change those guys' versions to the .NET Core 2.0 ones, but at this point I'm rewriting so much I'd much rather just use `mvc` template from the .NET 2.2 SDK. It looks to me like I have no such option? Do I really have to reinstall the 3.0 RC1 SDK and install the 2.2 one in order to be able to run `dotnet new mvc` and get the template which has NuGet packages and TFM for the 2.2 runtime?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ab345482-5752-6845-feee-52f0f06c620c
* Version Independent ID: 3613d4c0-be8e-c5a4-42f4-bec78ab99c75
* Content: [Select which .NET Core version to use - .NET Core](https://docs.microsoft.com/en-us/dotnet/core/versions/selection)
* Content Source: [docs/core/versions/selection.md](https://github.com/dotnet/docs/blob/master/docs/core/versions/selection.md)
* Product: **dotnet-core**
* GitHub Login: @Thraka
* Microsoft Alias: **adegeo**

**Comment by KevinCathcart** [comment]
It is entirely possible to do this, but the documentation appears to be pretty much non-existent.

First install the older version template pack you need. For MVC 2.2 it would be: 
```
dotnet new --install "Microsoft.DotNet.Web.ProjectTemplates.2.2"
```

Now you can specify a target framework for `dotnet new mvc` as follows:
```
dotnet new mvc --framework netcoreapp2.2
```

Hope this helps, at least until Microsoft update the documentation for this.

**Comment by TomasHubelbauer** [comment]
Thank you!

**Comment by mairaw** [comment]
I'll keep it open so we can check where updates are applicable.

**Comment by KevinCathcart** [comment]
Also, thanks to .net core 3 including a template search system, it is not required to actually know the name of the package to install.

Just blindly add the framework parameter and it will spit out the correct package to install, including the full command line to use. See:
```
root@9e3d403fdf2a:/# dotnet new mvc --framework netcoreapp2.2
Couldn't find an installed template that matches the input, searching online for one that does...
Matches from template source: NuGet
-----------------------------------
Template name "ASP.NET Core Web App (Model-View-Controller)" (mvc) from author "Microsoft" in pack Microsoft.DotNet.Web.ProjectTemplates.2.2
    To use this template, run the following command and try again:
        dotnet new -i Microsoft.DotNet.Web.ProjectTemplates.2.2::2.2.6
```

Hope this is useful in some way.

**Comment by Thraka** [comment]
After quite a bit of research we discovered that you really just need to install the templates for the specific version you want. However, with .NET Core directly (not ASP.NET Core) the templates for older versions aren't currently published to NuGet and can't be installed. I'm working with the product unit on a plan to address this.

**Comment by KevinCathcart** [comment]
As of 6 days ago, the .NET Core templates are now published on nuget.org, for 2.1 and newer. It is not immediately clear to me if 1.x and 2.0 templates were deliberately not published, or if the old beta versions published under @sayedihashimi’s account are blocking their upload.

I also want to point out that issues #3178, and #5961 are essentially also requesting this same documentation, and should probably be handled at the same time.

**Comment by Thraka** [comment]
Not sure where the ball was dropped originally, (i just think no one thought about using them in this way) but as part of my investigation on this issue I brought the fact that these missing nuget packages were a hurdle in addressing this scenario. So the team has been working on getting this packaged properly for NuGet and uploaded. I'm glad to hear it! I'll continue working on updating the documentation for this.

Thank you for your patience and help!

**Comment by Thraka** [comment]
@KevinCathcart Can you share where these packages are? I contacted the team and they said they're not published yet.

**Comment by KevinCathcart** [comment]
Strange.

https://www.nuget.org/packages/Microsoft.DotNet.Common.ProjectTemplates.2.1
https://www.nuget.org/packages/Microsoft.DotNet.Common.ProjectTemplates.2.2
https://www.nuget.org/packages/Microsoft.DotNet.Common.ProjectTemplates.3.0
https://www.nuget.org/packages/Microsoft.DotNet.Common.ProjectTemplates.3.1

all seem perfectly published to me.

The 2.0 and 1.x are not published (yet?).

Perhaps what the product team really means is that they are not fully published yet, and that they want to get the 2.0, 1.x, and ItemTemplates published before they consider this done? Or perhaps they want to get certain older versions published too? Not really clear to me.

At the very least those packages that I show above ought to be enough to validate that creating a .net core 2.1 app template from the latest 3.1 SDK works.

**Comment by Thraka** [comment]
Yeah, I'm wondering if as soon as the altered the project files to be compatible with NuGet (or something along these lines) this was incidentally automated.

I'll check with them. Thanks for the links.
